ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @module montage/ui/segmented-bar.reel
+ * @requires montage/ui/base/abstract-button.AbstractButton
+ * @requires montage/core/tree-controller.TreeController
+ */
+var AbstractButton = require("ui/base/abstract-button").AbstractButton;
+
+/**
+ * @class SegmentedBar
+ * @classdesc
+ * A segmented bar.
+ * Reference design: OS X storage space visualization found in iTunes or About This Mac
+ * Inheriting from AbstractButton because SegmentedBar needs its PressComposer behavior
+ *
+ * @extends AbstractButton
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes}
+ * for FlexBox usage
+ */
+exports.SegmentedBar = AbstractButton.specialize( /** @lends SegmentedBar.prototype # */ {
```

Comment:
> What if the value of isSelectionEnabled is changed while the SegmentedBar is in the DOM? Will the selection bindings be set or cleared accordingly? It doesn't seem like it.

There's no logic to account for changing `isSelectionEnabled` after instantiation. I was thinking about the possible use cases for SegmentedBar. It's highly unlikely someone would do that IMO.

I also didn't account for someone setting `data`, then setting `dataController` with different dataset, then go back and force on `data` and `dataController` with different datasets. Also too edge of a case IMO.

How edge must a case be before we decide to not account for it?